src/Revise.jl

```diff
@@ -405,6 +413,20 @@ function parse_module!(md::ModDict, ex::Expr, file::Symbol, mod::Module, path)
     newmod
 end
 
+function watch_files_via_dir(dirname)
+    watch_file(dirname)  # this will block until there is a modification
+    latestfiles = String[]
+    wf = watched_files[dirname]
+    for file in wf.trackedfiles
+        path = joinpath(dirname, file)
+        if mtime(path) > wf.timestamp
```

Comment:
If I change this line to `if mtime(path) >= floor(wf.timestamp)`, then all tests pass, even with `sleep(0.1)`. It might trigger a few needless checks into the files, but that should be rather minor.